Add concrete cryptosuites to deliverables.

msporny opened the pull request:
This PR implements a VCWG request from the 2021-12-15 telecon -- the request was to make it clear that the group intends to deliver concrete serializations of data integrity proofs (we are not just doing a meta data model). It was also requested to make it clear which cryptosuites and data integrity mechanisms were in scope for normative deliverables.

The JWP one is a bit out on a limb given that we don't have any implementations or a spec that goes along w/ the VC Data Model spec... but some felt it might be good to include it as an optional normative deliverable if the IETF RFCs materialize in time. We have working implementations, specs, and initial test suites for all the other normative deliverables.

This PR also updates the VC 1.0 spec references to VC 1.1 since that will be a reality by the time people get around to really analyzing this charter.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-wg-charter/pull/37.html" title="Last updated on Jan 13, 2022, 8:40 PM UTC (409f068)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-wg-charter/37/264504c...409f068.html" title="Last updated on Jan 13, 2022, 8:40 PM UTC (409f068)">Diff</a>

msporny committed:
Implement VCWG request from 2021-12-15 telecon.

iherman commented on index.html:
```diff
@@ -186,20 +186,20 @@ <h3>
+Exclusion period <b>began</b> 11 November 2021; Exclusion period <b>ended</b> 08 January 2021.
                   </p>
                   <p>
 <b>Other Charter:</b> <a href="https://www.w3.org/2017/vc/WG/charter.html"> https://www.w3.org/2017/vc/WG/charter.html</a>
```
By introducing 1.1 (which is fine) we may consider using the current WG charter reference (ie, https://www.w3.org/2020/12/verifiable-credentials-wg-charter.html) or both.

I am not 100% sure which one to take; maybe adding both is the safest bet.

msporny replied:
I'll use the latest charter ref, that's probably the easiest to understand.

-0.5 to using both, people would probably be confused by that... e.g., "Why are they listing two charters here? Are they both active? Only one of them?"

iherman replied:
Works for me.

msporny replied:
Done in https://github.com/w3c/vc-wg-charter/pull/37/commits/65f83eb97cd6c2d3cbf827186993e67426284379

TallTed commented on index.html:
```diff
@@ -186,20 +186,20 @@ <h3>
-<a href='https://www.w3.org/TR/2019/REC-vc-data-model-20191119/'>Verifiable Credentials Data Model 1.0</a>
-Exclusion period <b>began</b> 25 July 2019; Exclusion period <b>ended</b> 23 September 2019.
+<a href='https://www.w3.org/TR/2021/REC-vc-data-model-20211109/'>Verifiable Credentials Data Model 1.1</a>
+Exclusion period <b>began</b> 11 November 2021; Exclusion period <b>ended</b> 08 January 2021.
```
```suggestion
Exclusion period <b>began</b> 11 November 2021; Exclusion period <b>ended</b> 08 January 2022.
```
... else it ends before it began

TallTed commented on index.html:
```diff
@@ -212,10 +212,24 @@ <h3>
               <p>
-This specification defines how to express proofs of integrity, such as digital signatures or proofs of existence, for bounded documents such as verifiable credentials.
+This specification defines how to express proofs of integrity, such as digital
+signatures or proofs of existence, for bounded documents such as verifiable
```
```suggestion
signatures or proofs of existence, for bounded documents, such as verifiable
```
Could leave this comma out, IFF removing the one after `proofs of integrity`.

TallTed commented on index.html:
```diff
@@ -212,10 +212,24 @@ <h3>
+href="https://w3c.github.io/vc-data-model/#json-web-token">VC-JWT</a> and the <a
+href="https://w3c-ccg.github.io/lds-ed25519-2020/">Ed25519 Cryptosuite</a>.
+Concrete serializations might be provided for a <a
+href="https://w3c-ccg.github.io/ldp-bbs2020/">BBS+ Cryptosuite</a> if an <a
+href="https://identity.foundation/bbs-signature/">IETF BBS+ RFC</a> is published
+in time as well as VC-JWP if an <a
```
```suggestion
href="https://w3c-ccg.github.io/ldp-bbs2020/">BBS+ Cryptosuite</a>, if an <a
href="https://identity.foundation/bbs-signature/">IETF BBS+ RFC</a> is published
in time, as well as VC-JWP, if an <a
```

TallTed reviewed (changes requested):
Minor tweaks.

msporny committed:
Apply suggestions from @TallTed to cryptosuite deliverables.

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

msporny commented on index.html:
```diff
@@ -186,20 +186,20 @@ <h3>
+Exclusion period <b>began</b> 11 November 2021; Exclusion period <b>ended</b> 08 January 2021.
                   </p>
                   <p>
 <b>Other Charter:</b> <a href="https://www.w3.org/2017/vc/WG/charter.html"> https://www.w3.org/2017/vc/WG/charter.html</a>
```
```suggestion
<b>Other Charter:</b> <a href="https://www.w3.org/2020/12/verifiable-credentials-wg-charter.html"> https://www.w3.org/2020/12/verifiable-credentials-wg-charter.html</a>
```

msporny committed:
Update reference to latest VCWG charter.

msporny commented on index.html:
```diff
@@ -212,10 +212,24 @@ <h3>
-This specification defines how to express proofs of integrity, such as digital signatures or proofs of existence, for bounded documents such as verifiable credentials.
+This specification defines how to express proofs of integrity, such as digital
+signatures or proofs of existence, for bounded documents, such as verifiable
+credentials. Concrete serializations will be provided for <a
```
```suggestion
credentials. Concrete serializations will be provided based on <a
```

msporny commented on index.html:
```diff
@@ -212,10 +212,24 @@ <h3>
+credentials. Concrete serializations will be provided for <a
+href="https://w3c.github.io/vc-data-model/#json-web-token">VC-JWT</a> and the <a
+href="https://w3c-ccg.github.io/lds-ed25519-2020/">Ed25519 Cryptosuite</a>.
+Concrete serializations might be provided for a <a
```
```suggestion
Concrete serializations might be provided based on the <a
```

msporny reviewed (commented):
Minor rewording to not tie us directly to the input documents.

msporny committed:
Minor rewording to develop concrete cryptosuites based on input documents.

TallTed commented on index.html:
```diff
@@ -214,10 +214,10 @@ <h3>
-credentials. Concrete serializations will be provided for <a
+credentials. Concrete serializations will be provided based on <a
 href="https://w3c.github.io/vc-data-model/#json-web-token">VC-JWT</a> and the <a
 href="https://w3c-ccg.github.io/lds-ed25519-2020/">Ed25519 Cryptosuite</a>.
```
```suggestion
credentials. Concrete serializations based on <a
href="https://www.w3.org/TR/vc-data-model/#json-web-token">VC-JWT</a> and the <a
href="https://w3c-ccg.github.io/lds-ed25519-2020/">Ed25519 Cryptosuite</a> will
be provided.
```

iherman replied:
Actually... it would be more appropriate to refer to the TR document, ie, https://www.w3.org/TR/vc-data-model/#json-web-token

(I have not checked whether this is an issue elsewhere)

msporny replied:
I modified @TallTed's suggestion to include the TR URL.

TallTed commented on index.html:
```diff
@@ -214,10 +214,10 @@ <h3>
+Concrete serializations might be provided based on the <a
 href="https://w3c-ccg.github.io/ldp-bbs2020/">BBS+ Cryptosuite</a>, if an <a
 href="https://identity.foundation/bbs-signature/">IETF BBS+ RFC</a> is published
 in time, as well as VC-JWP, if an <a
```
```suggestion
Concrete serializations based on the <a
href="https://w3c-ccg.github.io/ldp-bbs2020/">BBS+ Cryptosuite</a> might be
provided, if an <a
href="https://identity.foundation/bbs-signature/">IETF BBS+ RFC</a> is published
in time, as well as based on VC-JWP, if an <a
```

TallTed replied:
The latter sentence might be better broken into two, even if they are worded almost identically.

msporny replied:
Yeah, I was on the fence about that as well. I kept it as the same sentence to try and convey "these two things are in the same category of "if an IETF RFC appears, we can standardize it."

TallTed replied:
I don't think the categorization of these two things is particularly important, because they're waiting on different RFCs, which are not tied together -- either RFC might be published, or both, or neither.  On that basis, I think it will actually be significantly clearer to split the sentence into two -- one for each serialization which might be provided, each dependent on publication of one RFC.

msporny replied:
Ok, agree with your logic @TallTed, I will make that change when I get to my next edit cycle.

msporny replied:
Done in https://github.com/w3c/vc-wg-charter/pull/37/commits/8abd5831d96ebe323ad5fc9c05faa89c62fa48e1.

peacekeeper commented:
Looks good, my only question is whether it would be forbidden to also work on other concrete suites that are not explicitly mentioned. Right now it says the WG **will** work on Ed25519, and **might** work on BBS+. Would the WG be allowed to optionally also work on others?

msporny commented:
> Looks good, my only question is whether it would be forbidden to also work on other concrete suites that are not explicitly mentioned. Right now it says the WG **will** work on Ed25519, and **might** work on BBS+. Would the WG be allowed to optionally also work on others?

Good point, we should say we might work on other cryptosuites that have WG buy-in. Although, that's usually a red flag for reviewers (open ended charter to develop normative documents are typically frowned upon). So, we'd have to point to some sort of draft if possible.

msporny committed:
Clarify cryptosuite work that is in scope.

* Final output specs might be different.
* We might do additional specs based on IETF RFCs.
* We plan to do one for Secp256k1

msporny commented:
> Would the WG be allowed to optionally also work on others?

@brentzundel and @peacekeeper, I updated the PR to make it clear that we can work on others as long as we're using IETF approved RFCs.

@rxgrant, I also added secp256k1.

We really should add secp*r1, but need a draft for that. I will do that in a separate PR.

I believe this PR addresses all input received in this PR and on the call and is ready to be merged.

iherman commented:
All approved, merging.

iherman commented:
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2022-01-12-vcwg#section3-1) on 2022-01-12

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 3.1. Add concrete cryptosuites to deliverables. (pr vc-wg-charter#37)

_See github pull request [vc-wg-charter#37](https://github.com/w3c/vc-wg-charter/pull/37)._




**Manu Sporny:** At a previous meeting we decided we wanted to move some cryptosuites into normative deliverables..  
… Data Integrity is not a concrete serialization, but Ed25519 and VC-JWT are.  
… This says we will work on concrete cryptosuites, and it lists some preliminary work that the WG could standardize.  
… Markus raised a good point, what if we want to standardize additional cryptosuites. Current language isn't clear on it.  
… Specifically, we don't directly list NIST curves like P-386, we probably should add some language.  

> *Ryan Grant:* +1 to language allowing too add curves.

**Manu Sporny:** Prefer to not pull in the PR until that language is in there.  

**Brent Zundel:** I think we could leave it a bit more open.  

**Ryan Grant:** Is there a requirement to focus only on NIST curves?.  

**Manu Sporny:** k1 curve potentially.  
… If we are vague, we might do normative things on all of it. This might be a red flag on charter review, since we're not very clear on concrete realizations we want to do..  
… Feedback may be we need to specify this exactly..  
… W3C members don't like very open charters.  
… That's the challenge with making it more general. If we can't point to a draft spec, people might push back on it..  
… Easiest thing to do would be to point to draft spec. I think there is one for k1..  

> *Juan Caballero:* it may be helpful. i believe it's MIT if I remember correctly.

**Manu Sporny:** Before charter goes out for review, someone could create a cryptosuite.  
… Safest thing to do would be, if you want a particular type of cryposuite to be standardized, create a draft spec for it that we can cite in the charter..  

> *Markus Sabadello:* [https://w3c-ccg.github.io/lds-ecdsa-secp256k1-2019/](https://w3c-ccg.github.io/lds-ecdsa-secp256k1-2019/) ?.

> *Ryan Grant:* makes sense here.

**Brent Zundel:** How harmful would it be to include a line "an others as determined by the WG".  

**Manu Sporny:** We can do that, I would expect that to result in a Formal Objection.  
… We could try.  
… If we are tied to it, we probably should be very specific.  

**Brent Zundel:** So we hold off on merging this PR.  
… Make comments on the PR if you want additional cryptosuites.  

**Manu Sporny:** If someone wants a NIST curve, we could put together a cryptosuite.  

**Brent Zundel:** Other comments or questions?.  

</details>
